rasa_core/validator.py

```diff
@@ -0,0 +1,394 @@
+import logging
+import os
+from jsonschema import validate
+import yaml
+from os import listdir
+from os.path import isfile, join
+import argparse
+from rasa_core import utils
+
+
+logger = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument(
+    '--domain', '-d',
+    type=str, default='domain.yml',
+    help='Path for the domain file'
+)
+
+parser.add_argument(
+    '--stories', '-s',
+    type=str, default='data/stories.md',
+    help='Path for the stories file or directory'
+)
+
+parser.add_argument(
+    '--intents', '-i',
+    type=str, default='data/intents.md',
+    help='Path for the intents file or directory'
+)
+
+parser.add_argument(
+    '--warnings', '-w',
+    action='store_true',
+    default=False,
+    help='Run script with warings'
+)
+
+parser.add_argument(
+    '--no_validate_intents', action='store_true', default=False,
+    help='Skips validations to intents'
+)
+
+parser.add_argument(
+    '--no_validate_utters', action='store_true', default=False,
+    help='Skips validations to utters'
+)
+
+parser.add_argument(
+    '--no_validate_domain', action='store_true', default=False,
+    help='Skips validations to domain'
+)
+
+
+class Validator:
+    domain = ''
+    intents = []
+    stories = []
+    valid_intents = []
+    valid_utters = []
+
+    def __init__(self, domain="domain.yml", intents="data/nlu.md",
+                 stories="data/stories.md"):
+
+        # Saving domain file
+        if os.path.exists(domain):
+            self.domain = domain
+        else:
+            logger.error("The domain file was not found")
+
+        # Saving intents files
+        if os.path.isfile(intents) and os.path.exists(intents):
+            self.intents.append(intents)
+
+        elif os.path.isdir(intents):
+            if not intents.endswith('/'):
+                intents += '/'
+
+            intent_files = [f for f in listdir(intents)
+                            if isfile(join(intents, f))]
+            for file in intent_files:
+                self.intents.append(intents + file)
+
+        else:
+            logger.error("The intents file was not found")
+
+        # Saving stories files
+        if os.path.isfile(stories) and os.path.exists(stories):
+            self.stories.append(stories)
+
+        elif os.path.isdir(stories):
+            if not stories.endswith('/'):
+                stories += '/'
+
+            stories_files = [f for f in listdir(stories)
+                             if isfile(join(stories, f))]
+            for file in stories_files:
+                self.stories.append(stories + file)
+
+        else:
+            logger.error("The stories file was not found")
```

Comment:
Need a little refactoring, could you make a function called `from_paths(intent_filepath, story_filepath, domain_filepath)` which performs the above checks `for path in paths`? (i.e. for intents, stories and domain)

This function should then return the class (see [here](https://github.com/RasaHQ/rasa_core/blob/master/rasa_core/agent.py#L267) for an example)

Reply:
The refactoring was made [here](https://github.com/lappis-unb/rasa_core/blob/24d89908dbbab1081b9195949fb6ff7da05561e9/rasa_core/validator.py#L71)
It returns a cls that creates the Validator object [here](https://github.com/lappis-unb/rasa_core/blob/24d89908dbbab1081b9195949fb6ff7da05561e9/rasa_core/validator.py#L107)
The classes [StoryFileReader](
https://github.com/RasaHQ/rasa_core/blob/master/rasa_core/training/dsl.py#L146), [TrainingData](
https://github.com/RasaHQ/rasa_nlu/blob/master/rasa_nlu/training_data/loading.py#L43) and [Domain](
https://github.com/RasaHQ/rasa_core/blob/master/rasa_core/domain.py#L76) return a processed json when opening the file. The goal of the validator is to verify the structure of the files before running these classes to assure it will go well. My solution was to save the path for the file and open them to verify if it has the correct structure. Is there any update to be made about it?